src/native/corehost/hostmisc/pal.unix.cpp

```diff
@@ -424,35 +417,68 @@ bool pal::get_dotnet_self_registered_dir(pal::string_t* recv)
     FILE* install_location_file = pal::file_open(install_location_file_path, "r");
     if (install_location_file == nullptr)
     {
-        trace::verbose(_X("The install_location file failed to open."));
+        trace::error(_X("The install_location file failed to open."));
         return false;
     }
 
-    bool result = false;
-
-    char buf[PATH_MAX];
-    char* install_location = fgets(buf, sizeof(buf), install_location_file);
-    if (install_location != nullptr)
+    char buffer[PATH_MAX];
+    pal::string_t install_location, architecture_install_location;
+    bool is_first_line = true;
+    while (fgets(buffer, sizeof(buffer), install_location_file))
     {
-        size_t len = pal::strlen(install_location);
-
+        size_t len = strlen(buffer);
         // fgets includes the newline character in the string - so remove it.
-        if (len > 0 && len < PATH_MAX && install_location[len - 1] == '\n')
+        if (len > 0 && len < PATH_MAX && buffer[len - 1] == '\n')
         {
-            install_location[len - 1] = '\0';
+            buffer[len - 1] = '\0';
         }
 
-        trace::verbose(_X("Using install location '%s'."), install_location);
+        char* pDelimiter = strchr(buffer, '=');
+        if (pDelimiter == nullptr && len > 0)
+        {
+            if (is_first_line)
+            {
+                install_location = pal::string_t(buffer);
+                trace::verbose(_X("Found install location path '%s'."), install_location.c_str());
+            }
+            else
+            {
+                trace::error(_X("Only the first line in '%s' may not have an architecture prefix."), install_location_file_path.c_str());
+            }
+
+            is_first_line = false;
+            continue;
+        }
+
+        auto delimiter_index = (int)(strlen(buffer) - strlen(pDelimiter));
+        pal::string_t arch_prefix = pal::string_t(buffer).substr(0, delimiter_index);
+        pal::string_t path_to_location = pal::string_t(buffer).substr(delimiter_index + 1);
```

Comment:
This will create the string wrapper twice (probably also make two copies). As noted above let's make fgets which returns `pal::string_t` and then the rest can operate on it directly.